feat: show values above graphs on perf summary

alxu-sentry opened the pull request:
While we're here, also do some UserStats optimizations and make the sidebar
spacing more consistent.

<img width="344" alt="Screen Shot 2021-03-08 at 8 56 03 AM" src="https://user-images.githubusercontent.com/78757344/110330729-208c3100-7fec-11eb-8d49-d98ff263b787.png">

alxu-sentry committed:
feat: show values above graphs on perf summary

While we're here, also do some UserStats optimizations and make the sidebar
spacing more consistent.

alxu-sentry commented on src/sentry/static/sentry/app/views/performance/transactionSummary/sidebarCharts.tsx:
```diff
@@ -34,9 +35,10 @@ type Props = ReactRouter.WithRouterProps & {
   organization: LightWeightOrganization;
   location: Location;
   eventView: EventView;
+  totals: Record<string, number> | null;
```
`| null` isn't useful now but will be after #24298.

alxu-sentry commented on tests/js/spec/views/performance/transactionSummary.spec.jsx:
```diff
@@ -91,18 +91,22 @@ describe('Performance > TransactionSummary', function () {
             count: 'number',
-            apdex_300: 'number',
-            user_misery_300: 'number',
+            apdex_400: 'number',
```
I think these numbers were wrong the whole time but nobody noticed.

alxu-sentry committed:
restore the bottom of the sidebar graphs

doralchan reviewed (approved):
I dunno why part of the y-axis is cropped in your screenshot but it looks fine when I look at it locally. ¯\_(ツ)_/¯

Zylphrex commented on src/sentry/static/sentry/app/views/performance/transactionSummary/userStats.tsx:
```diff
@@ -54,9 +55,7 @@ function UserStats({eventView, totals, location, organization, transactionName}:
-    const apdexKey = `apdex_${threshold}`;
-    const formatter = getFieldRenderer(apdexKey, {[apdexKey]: 'number'});
-    apdex = formatter(totals, {organization, location});
+    apdex = totals[`apdex_${threshold}`].toFixed(4);
```
Why is this using `.toFixed(4)` over the `formatter` here?

alxu-sentry replied:
invoking the formatter is overly verbose and the only thing it really does is `text-align: right` which we don't want here anyways.

Zylphrex replied:
can we use [formatFloat](https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/utils/formatters.tsx#L180-L184) instead to be consistent with how we format numbers in (most of) the app.

alxu-sentry replied:
actually I think more stuff uses toFixed than formatFloat, but either way I want to kill formatFloat entirely because it's implemented inefficiently and functions very unexpectedly with large numbers or large `places`:

```
> formatFloat(1e11, 10)
1e-10
```

but i'm worried that something could possibly actually require this behavior.

Zylphrex replied:
I mention using `formatFloat` because `.toFixed` has some gotchas to watch out for due to the precision of floating-point numbers see [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed#using_tofixed).

But mainly, the concern is to make sure we format `apdex` the same way in all parts of the application 🙂 (i.e. we don't format it with 4 decimal places here and 2 elsewhere)

alxu-sentry replied:
that's true, but i don't understand how the warning there is related to toFixed.

i did realize that the two aren't actually the same though: `toFixed` adds trailing zeroes, `formatFloat` doesn't. i kinda feel like trailing zeroes is more accurate, but it's better to be consistent.

alxu-sentry commented:
> I dunno why part of the y-axis is cropped in your screenshot but it looks fine when I look at it locally. ¯_(ツ)_/¯

i fixed it in 99dc3396deff9bb5178bde45663bedfeda562255 but didn't update the screenshot. sorry for any confusion

alxu-sentry committed:
import styled from @emotion/styled

Suggested-by: Billy Vong <billyvg@users.noreply.github.com>

alxu-sentry committed:
switch back to fieldrenderer

it's a lot of boilerplate but it's the only place where the constant 4 is
defined.

Zylphrex commented on src/sentry/static/sentry/app/views/performance/transactionSummary/sidebarCharts.tsx:
```diff
@@ -141,35 +143,57 @@ function SidebarCharts({api, eventView, organization, router}: Props) {
+          />
+        </ChartTitle>
+        {totals ? (
+          <ChartValue>{tct('[tpm] tpm', {tpm: totals.tpm.toFixed(0)})}</ChartValue>
```
We typically format TPM with up to 4 decimal places. You can see an example of this on the performance landing page.

alxu-sentry replied:
I'd argue that it's a bit dubious whether we should, but I'll change it to 4 for now for consistency.

alxu-sentry committed:
adjust sidebar charts formatting

1. formatFloat doesn't pad with zeroes
2. tpm to 4 places for consistency

alxu-sentry committed:
Merge branch 'master' into transaction-summary-graph-numbers

Zylphrex reviewed (approved):
LGTM
